### ISSUE TITLE
Quit hangs

### ISSUE THREAD

**Issue by psznm** [logdna/tail-file-node#38]
I prepared simple script to demonstrate. If you run the script, you can see that after calling quit on the tail instance, it will never quit and will hang script until it is garbage collected.

Also honest question. Why does actually quit try to read one more time? When I tell it to quit, I do not want any more data, I want it to quit now...?

```
const TailFile = require('@logdna/tail-file')
const fs = require('fs');
const TMP_FILE = '/tmp/logDnaTailFileTest';
const add5 = () => fs.writeFileSync(TMP_FILE, '12345', {flag: 'a'});
const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));


const main = async () => {

    console.error("Preparing initial data file");
    if (fs.existsSync(TMP_FILE)) {
        fs.unlinkSync(TMP_FILE)
    }
    add5();

    const tail = new TailFile(TMP_FILE, {highWaterMark: 7});
    const preventGarbageCollectionTimeout = setTimeout(() => {
        console.error(`Losing reference to tail. It will now be garbage collected. Along with the hanging await. ${tail}`);
    }, 10000)

    console.error("Waiting until tail loads some data");
    tail.start()
    await sleep(10);

    console.error("Adding additional data and simultaneously trying to quit.")
    add5();
    add5();
    // Never quits. Hangs until garbage collected
    tail.quit().then(() => {
        clearTimeout(preventGarbageCollectionTimeout);
        console.error("Tail has quit.")
    })

    // // quits immediately
    // tail.quit('err').catch(() => {
    //     clearTimeout(preventGarbageCollectionTimeout);
    //     console.error("Tail has errored.")
    // });
}
main();
```

**Comment by darinspivey** [comment]
Wow, REALLY sorry for the delay in response to this!  This issue simply got overlooked and forgotten.

So, there's 2 things talked about in this issue:
1) quit hangs
2) Why does `quit` read one more time?

First, let's talk about the second thing.  The simple reason is that, in high throughput, the read buffer may be full and it seemed prudent to try to read whatever may be there.  Our company uses this to process logs, so not getting those lines is technically data loss.  If a user is looking for certain information, and immediately wants to stop, we can make that a new parameter to the quit function, eg. `.quit({immediate: true})`.  It should be noted that this extra read should be very fast, but we understand that if data is being saved, there could be extra lines added unexpectedly.

For the first point, the example code you submitted has a problem.  The file stream must be put into "flowing mode" by either a `.pipe()` or a `.on('data')` event listener.  By not doing this, nothing is able to read, and the extra read is causing the hang.  When using the examples, if we add a `readline` interface, the stream will be put in flowing mode with a pipe behind the scenes.  When doing that, the code exits immediately.

That being said, we'll add the feature to exit immediately if requested, although it's not clear if setting up `TailFile` without flowing mode is actually a use case (for the hang issue), but extra data is a valid concern.

Corrected code:

```js
'use strict'

const TailFile = require('@logdna/tail-file')
const readline = require('readline')
const fs = require('fs');
const TMP_FILE = '/tmp/logDnaTailFileTest';
const add5 = () => fs.writeFileSync(TMP_FILE, '12345', {flag: 'a'});
const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));


const main = async () => {

    console.error("Preparing initial data file");
    if (fs.existsSync(TMP_FILE)) {
        fs.unlinkSync(TMP_FILE)
    }
    add5();

    const tail = new TailFile(TMP_FILE, {highWaterMark: 7});
    const preventGarbageCollectionTimeout = setTimeout(() => {
        console.error(`Losing reference to tail. It will now be garbage collected. Along with the hanging await. ${tail}`);
    }, 10000)

    const linesplitter = readline.createInterface({
      input: tail
    })

    linesplitter.on('line', (line) => {
      console.log(line)
    })

    console.error("Waiting until tail loads some data");
    tail.start()
    await sleep(10);

    console.error("Adding additional data and simultaneously trying to quit.")
    add5();
    add5();
    // Never quits. Hangs until garbage collected
    tail.quit().then(() => {
        clearTimeout(preventGarbageCollectionTimeout);
        console.error("Tail has quit.")
    })

    // // quits immediately
    // tail.quit('err').catch(() => {
    //     clearTimeout(preventGarbageCollectionTimeout);
    //     console.error("Tail has errored.")
    // });
}
main();

```

**Comment by psznm** [comment]
Hello, thank you for your response. My scenario was pipeline TailFile > CustomTransform > process.stdout. I think the issue was happening where if process.stdout was closed, both TailFile and CustomTransform were also immediately closed and program was expecting to quit immediately, which did not happen, sometimes for several seconds. 

At the time I went searching for issues that could be causing this, and I found this, so I was blaming TailFile, but eventually I found out I accidentally instantiated TailFile twice and lost reference to the first instance immediately. My issue must have been caused by the first instance of TailFile still reading. 

After removing the extraneous instance of TailFile, my issue with hanging program has been resolved and thus this github issue is no longer relevant to me.
